### PR TITLE
Introduce general error handling (and other stuff)

### LINKED ISSUE #6: Scripts goes on if a mission-critical command fails
As a general case of #4, if any mission-critical command within the `./make_cluster` (say, an `ssh` connection) fails, execution flow doesn't change despite the fact that a command failed.

What's needed here is to perform a check for most important commands like `ssh` and halt execution if an error is detected. I'm already on it and will provide a proper PR to fix this asap.

### PR DESCRIPTION
**Pull request by bissim**
Fixes #6

As stated in #6, now every mission-critical command in `make_cluster.sh` script gets checked and if it fails execution is halted; it is a general case of the fix provided in PR #5 for issue #4.

I also reviewed English sentences along the scripts and README, [...]; moreover, I took the chance to make README  a bit more clearer, like removing a point in [_Dependencies_](https://github.com/isislab-unisa/aws-build-cluster-script/blob/master/README.md#dependencies) paragraph that is actually an anticipation of what's required in next paragraph and not an actual operative instruction (that would've led users to confusion).

Minor review in the sense to fix English has been apported to `state_cluster.sh` script as well.

`VERSION` has been bumped to 1.2.2 to eventually trigger any user of 1.2.0 and 1.2.1 to download the new version.

### CODE REVIEW

**Comment by bissim** [comment]
A subtlety: in f3031d7 I also put version information in splash screen.

File: README.md [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 
-To use this script you must have AWS CLI and jq 
+To use this script you must have AWS CLI and `jq` installed 
 
-1. Install the **AWS CLI**, see [here](https://docs.aws.amazon.com/cli/latest/userguide/installing.html)
-   If you don't have it, you must install Python.
-   Test if it works with the following command:
+1. Install the **AWS CLI**, see [official documentation](https://docs.aws.amazon.com/cli/latest/userguide/installing.html)
+   For AWS CLI v1, if you don't have it, you must install Python; for v2 all required tools are self-contained.
+   Test if it works by launching the following command:
```
Comment (bissim) [inline README.md]: Here, an user may be misled if it's using AWS CLI v2 that integrates Python and is therefore not needed in the environment. Quoting from [official source](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html):
> For AWS CLI version 2, it doesn't matter if you have Python installed and if you do, it doesn't matter which version. AWS CLI version 2 uses only the embedded version of Python (and any other dependencies) that is included in the installer.

File: README.md [hunk 2]
```diff
@@ -79,27 +79,21 @@ To use this script you must have AWS CLI and jq
 	$ chmod +x make_cluster.sh state_cluster.sh
 	```
 
-4. Choose the type of machine to use in the cluster
-
-   Now you can choose what kind of machine: you must take note of the code name.
-   See the [catalog](https://aws.amazon.com/ec2/instance-types/)
-
-
```
Comment (bissim) [inline README.md]: This whole point is actually a requirement for _Create a cluster_ paragraph: that's the reason the catalog link has been moved to line 88.

File: README.md [hunk 3]
```diff
@@ -237,23 +229,25 @@ This command should show a json response with the state of all cluster instances
...
 
-The following table shows the other results of the tests performed on different OS:
+The following table shows the other results of the tests performed on different OSes:
 
 |      AMI     |                           AMI NAME                          | OS TYPE | INSTANCE TYPE | RESULT |
-|:------------:|:-----------------------------------------------------------:|:-------:|:-------------:|:------:|
-| ami-52a0c53b |           starcluster-base-ubuntu-12.04-x86_64-hvm          |  Ubuntu |    T2.Micro   |  WORK  |
-| ami-116d857a |         debian-jessie-amd64-hvm-2015-06-07-12-27-ebs        |  Debian |    T2.Micro   |  WORK  |
-| ami-00035c7b | Fedora-Atomic-25-20170727.0.x86_64-us-east-1-HVM-standard-0 |  Fedora |    T2.Micro   |  WORK  |
+|:------------:|:-----------------------------------------------------------:|:-------:|:-------------:|:-------:|
```
Comment (bissim) [inline README.md]: Whoops, I forgot two dashes here under `INSTANCE TYPE`.

**Comment by labgua** [comment]
> Fixes #6
> 
> As stated in #6, now every mission-critical command in `make_cluster.sh` script gets checked and if it fails execution is halted; it is a general case of the fix provided in PR #5 for issue #4.
> 
> I also reviewed English sentences along the scripts and README, since @labgua is not very fond with English (eh Sergio?); moreover, I took the chance to make README a bit more clearer, like removing a point in [_Dependencies_](https://github.com/isislab-unisa/aws-build-cluster-script/blob/master/README.md#dependencies) paragraph that is actually an anticipation of what's required in next paragraph and not an actual operative instruction (that would've led users to confusion).
> 
> Minor review in the sense to fix English has been apported to `state_cluster.sh` script as well.
> 
> `VERSION` has been bumped to 1.2.2 to eventually trigger any user of 1.2.0 and 1.2.1 to download the new version.

Salve @bissim 
La ringrazio per il suo contributo ma nessuno le chiede di aiutarci e nessuno le da il diritto di giudicare: se le và posti pure la tua soluzione.
Se non capisce cosa c'è scritto in questo commento le do un consiglio: usi google traduttore... sempre ammesso che sia questo il suo problema.

Here the link for the translation https://tinyurl.com/translbissim

**Comment by bissim** [comment]
After a brief review of my message (and a little bit of search in my dictionary), I came up to the source of this misunderstanding in my use of the expression _to be fond of_ that has an actual stronger meaning than the one I intended (still, it strangely gets translated nicely by Google Translator).
The meaning I intended was 'not to feel at one's ease'.

That being said, not feeling at one's ease with English is not something one should be ashamed of or feel guilt for, nor this is something I can feel responsible for in any degree or in any other way.
This is my last stance over this matter.

So, it's up to you to decide whether merge or reject this PR; I'm fine with whatever you decide.

**Comment by labgua** [comment]
Thanks a lot for your contribution.
I'll return the favour.

**Pull request merged**